source/adapters/cuda/event.cpp

```diff
@@ -129,6 +129,18 @@ ur_result_t ur_event_handle_t_::record() {
   return Result;
 }
 
+ur_result_t ur_event_handle_t_::make_end_event_same_as_start() {
+  if (isRecorded() || !isStarted()) {
+    return UR_RESULT_ERROR_INVALID_EVENT;
+  }
+  UR_ASSERT(Queue, UR_RESULT_ERROR_INVALID_QUEUE);
+
+  EvEnd = EvStart;
```

Comment:
Just a knit, but this function is and will always be called for events that are only visible inside the scope of another function (i.e., non "thread-escaping") ?
At least that's the case for urEnqueueTimestampRecordingExp if I see it correctly.
Otherwise, you might want to consider using mutexes at this point.

Reply:
This function is only called internally after a call to `start()` so there ought to never be any contention. For clarity I may combine this into a single call to `start`

Reply:
Have combined. See new diff.